src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client.py

```diff
@@ -251,4 +325,18 @@ def _run(args: argparse.Namespace) -> None:
         file_handler = logging.FileHandler(args.log_file, mode='a')
         file_handler.setFormatter(formatter)
         logger.addHandler(file_handler)
-    _run(args)
+    methods = args.rpc.split(",")
+    if set(methods) - set(_SUPPORTED_METHODS):
+        raise ValueError("--rpc supported methods: {}".format(
+            ", ".join(_SUPPORTED_METHODS)))
+    per_method_metadata = collections.defaultdict(list)
+    metadata = args.metadata.split(",") if args.metadata else []
+    for metadatum in metadata:
+        elems = metadatum.split(":")
+        if len(elems) != 3:
+            raise ValueError(
+                f"'{metadatum}' was not in the form 'METHOD:KEY:VALUE'")
+        if elems[0] not in _SUPPORTED_METHODS:
+            raise ValueError(f"Unrecognized method '{elems[0]}'")
+        per_method_metadata[elems[0]].append((elems[1], elems[2]))
```

Comment:
Agree. Guess I've been spending too much time in the RCTH codebase, where I've been asked to get rid of helper methods because they were "unnecessary". 😅

Reply:
Alternatively, we could add a bit comment to spatially separate each parts. I'm okay with either option.